src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.Workable;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> listMachines;
+        if (type.equals(Bulldozer.class)) {
+            listMachines = new BulldozerProducer().get();
+        } else if (type.equals(Excavator.class)) {
+            listMachines = new ExcavatorProducer().get();
+        } else if (type.equals(Truck.class)) {
+            listMachines = new TruckProducer().get();
```

Comment:
The `TruckProducer` class should be parameterized as `TruckProducer implements MachineProducer<Truck>` to ensure type safety and prevent the creation of invalid implementations.